packages/chart/src/ChartModel.ts

```diff
@@ -24,122 +29,131 @@ class ChartModel {
 
   constructor() {
     this.listeners = [];
-    this.formatter = null;
-    this.rect = null;
     this.isDownsamplingDisabled = false;
-    this.title = null;
   }
 
-  getData() {
+  listeners: ((event: ChartEvent) => void)[];
+
+  formatter?: Formatter;
+
+  rect?: DOMRect;
+
+  isDownsamplingDisabled: boolean;
+
+  title?: string;
+
+  getData(): [] {
```

Comment:
Need to have a type for the array returned - right now this is always specifying an empty array, which is why `FigureChartModel` doesn't conform (and you're having to specify `ChartModel | FigureChartModel` everywhere instead of just `ChartModel`).

You can just put `unknown[]` for now, unless you have a more specific type (I think we can fill that in when converting `FigureChartModel` though.
```suggestion
  getData(): unknown[] {
```